### PR TITLE
aws_secret - Support purge_tags

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

aws_secret currently defaults to purging all tags (even if tags isn't specified), this is a little aggressive.

1) Add purge_tags parameter
2) Only purge tags if `tags: {}` is set (rather than when `tags is None`

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

aws_secret

##### ADDITIONAL INFORMATION

Related to #1146

### CODE REVIEW

File: plugins/modules/aws_secret.py [hunk 1]
```diff
@@ -62,8 +62,15 @@
     version_added: 3.1.0
   tags:
     description:
-    - Specifies a list of user-defined tags that are attached to the secret.
+    - Specifies a dictionary of user-defined tags that are attached to the secret.
+    - To remove all tags set I(tags={}) and I(purge_tags=true).
     type: dict
+  purge_tags:
```
Comment (markuman) [inline plugins/modules/aws_secret.py:68]:
```suggestion
  purge_tags:
    version_added: 4.0.0
```

**Comment by tremble** [comment]
While the integration tests could be further expanded for the non-tag parameters, I'm going to leave them as-is for now.  If someone makes other changes to other parameters then we can get them to expand them for that parameter.

File: tests/integration/targets/aws_secret/tasks/basic.yml [hunk 2]
```diff
@@ -1,112 +1,573 @@
...
+    register: result
+
   - name: assert correct keys are returned
     assert:
       that:
         - result.changed
-        - result.arn is not none
-        - result.name is not none
-        - result.tags is not none
-        - result.version_ids_to_stages is not none
```
Comment (tremble) [inline tests/integration/targets/aws_secret/tasks/basic.yml:43]: This wasn't actually doing what the author thought it was doing.  The values were actually all in `result.secret`

File: plugins/modules/aws_secret.py [hunk 3]
```diff
@@ -139,6 +153,29 @@
       returned: always
       type: dict
       sample: { "dc1ed59b-6d8e-4450-8b41-536dfe4600a9": [ "AWSCURRENT" ] }
+    tags:
```
Comment (alinabuzachis) [inline plugins/modules/aws_secret.py:156]: Don't we also need to add `version_added` also for this returned parameter (even it's not new)?

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**